src/openforms/appointments/contrib/jcc/plugin.py

```diff
@@ -0,0 +1,183 @@
+
+        if not start_at:
+            start_at = date.today()
+        if not end_at:
+            end_at = start_at + timedelta(days=14)
+
+        try:
+            max_end_date = self.client.service.getGovLatestPlanDate(
+                productId=product_ids
+            )
+            if end_at > max_end_date:
+                end_at = max_end_date
+
+            days = self.client.service.getGovAvailableDays(
+                locationID=location.identifier,
+                productID=product_ids,
+                startDate=start_at,
+                endDate=end_at,
+                appDuration=0,
+            )
+            return days
+        except Exception as e:
+            logger.exception(e)
+            return []
+
+    def get_times(
+        self, products: list, location: AppointmentLocation, day: date
+    ) -> list:
+        product_ids = squash_ids(products)
+
+        try:
+            times = self.client.service.getGovAvailableTimesPerDay(
+                date=day,
+                productID=product_ids,
+                locationID=location.identifier,
+                appDuration=0,
+            )
+            return times
+        except Exception as e:
+            logger.exception(e)
+            return []
+
+    def create_appointment(
+        self,
+        products: list,
+        location: AppointmentLocation,
+        start_at: datetime,
+        client: AppointmentClient,
+        remarks: str = None,
+    ) -> str:
+        product_ids = squash_ids(products)
+
+        try:
+            factory = self.client.type_factory("http://www.genericCBS.org/GenericCBS/")
+            appointment_details = factory.AppointmentDetailsType(
+                locationID=location.identifier,
+                productID=product_ids,
+                clientLastName=client.last_name,
+                clientDateOfBirth=client.birthdate,
+                appStartTime=start_at,
+                appEndTime=start_at,  # Needed but unused by the service.
+                isClientVerified=False,
+                # Optional fields.
+                # These might be needed. Depends on `GetRequiredClientFields`
+                #
+                # clientID=bsn,
+                # clientSex="M/F",
+                # clientInitials="",
+                # clientAddress="",
+                # clientPostalCode="",
+                # clientCity="",
+                # clientCountry="",
+                # clientTel="",
+                # clientMail="",
+                appointmentDesc=remarks,
+                # caseID": "",
+            )
+
+            result = self.client.service.bookGovAppointment(
+                appDetail=appointment_details, fields=[]
+            )
+
+            if result["updateStatus"] == 0:
+                return result["appID"]
+            else:
+                raise Exception(
+                    "Could not create appointment (updateStatus=%s)",
+                    result["updateStatus"],
+                )
```

Comment:
Please use a specific `Exception` subclass for this so we can specifically catch those exceptions.

Reply:
Introduced `AppointmentException`.